static/app/views/detectors/hooks/useDeleteDetectorsMutation.tsx

```diff
@@ -0,0 +1,39 @@
+import {addErrorMessage, addSuccessMessage} from 'sentry/actionCreators/indicator';
+import {t} from 'sentry/locale';
+import {useMutation, useQueryClient} from 'sentry/utils/queryClient';
+import type RequestError from 'sentry/utils/requestError/requestError';
+import useApi from 'sentry/utils/useApi';
+import useOrganization from 'sentry/utils/useOrganization';
+
+/** Bulk delete detectors */
+export function useDeleteDetectorsMutation() {
+  const org = useOrganization();
+  const api = useApi({persistInFlight: true});
+  const queryClient = useQueryClient();
+
+  return useMutation<
+    void,
+    RequestError,
+    {ids?: string[]; projects?: number[]; query?: string}
+  >({
+    mutationFn: params => {
+      return api.requestPromise(`/organizations/${org.slug}/detectors/`, {
+        method: 'DELETE',
+        query: {
+          id: params.ids,
+          query: params.query,
+          project: params.projects,
+        },
+      });
+    },
+    onSuccess: () => {
+      queryClient.invalidateQueries({
+        queryKey: [`/organizations/${org.slug}/detectors/`],
+      });
+      addSuccessMessage(t('Monitors deleted'));
+    },
+    onError: error => {
+      addErrorMessage(t('Unable to delete monitors: %s', error.message));
```

Comment:
`error.message` isn't very useful - if the endpoint is returning text that you want to display you'll need to dig in a little more

<img width="563" height="75" alt="Image" src="https://github.com/user-attachments/assets/edc9abb6-a9fa-460b-bd89-99e2802eae5e" />

Reply:
updated this to use `error.responseText` for more detail https://github.com/getsentry/sentry/pull/97814/commits/8fb1f14eb6a791f1ab99d0ad056b850fce9eb62f